test/e2e/apimachinery/namespace.go

```diff
@@ -575,6 +559,20 @@ func ensurePodsAreRemovedFirstInOrderedNamespaceDeletion(ctx context.Context, f
 			}
 			return true, nil
 		}))
+	ginkgo.By("the pod should be deleted before processing deletion for other resources")
+	framework.ExpectNoError(wait.PollUntilContextTimeout(ctx, 2*time.Second, 60*time.Second, true,
+		func(ctx context.Context) (bool, error) {
+			_, err = f.ClientSet.CoreV1().ConfigMaps(nsName).Get(ctx, configMapName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "configmap %q should still exist in namespace %q", configMapName, nsName)
+			// the pod should exist and has a deletionTimestamp set
+			pod, err = f.ClientSet.CoreV1().Pods(nsName).Get(ctx, pod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to get pod %q in namespace %q", pod.Name, nsName)
+			if pod.DeletionTimestamp == nil {
+				framework.Logf("Pod %q in namespace %q does not yet have a metadata.deletionTimestamp set, retrying...", pod.Name, nsName)
+				return false, nil
+			}
+			return true, nil
+		}))
```

Comment:
this error makes it more confusing when it fails, this should be easier to read this way
```suggestion
	err := wait.PollUntilContextTimeout(ctx, 2*time.Second, 60*time.Second, true,
		func(ctx context.Context) (bool, error) {
			_, err = f.ClientSet.CoreV1().ConfigMaps(nsName).Get(ctx, configMapName, metav1.GetOptions{})
			framework.ExpectNoError(err, "configmap %q should still exist in namespace %q", configMapName, nsName)
			// the pod should exist and has a deletionTimestamp set
			pod, err = f.ClientSet.CoreV1().Pods(nsName).Get(ctx, pod.Name, metav1.GetOptions{})
			framework.ExpectNoError(err, "failed to get pod %q in namespace %q", pod.Name, nsName)
			if pod.DeletionTimestamp == nil {
				framework.Logf("Pod %q in namespace %q does not yet have a metadata.deletionTimestamp set, retrying...", pod.Name, nsName)
				return false, nil
			}
			return true, nil
		})
		if err != nil {
		       framework.Failf("pod %s was not deleted before the configmap %s: %v", pod.Nmae, configMapName, err) 
		    }
```

Reply:
Done